zerver/lib/users.py

```diff
@@ -65,6 +65,19 @@ def check_full_name(full_name_raw: str) -> str:
     # ban them.
     if re.search(r"\|\d+$", full_name_raw):
         raise JsonableError(_("Invalid format!"))
+
+    if require_unique_names(realm):
+        normalized_user_full_name = unicodedata.normalize("NFKC", full_name)
+        realm_users_names = UserProfile.objects.filter(realm=realm).values_list(
+            "full_name", flat=True
+        )
```

Comment:
I'm not sure if there's a better way to do this other than taking all the names of realm into a list and normalizing them, considering possibility of large number of users in realm. I tried implementing normalizing within query itself using the [`Func()`](https://docs.djangoproject.com/en/5.0/ref/models/expressions/#func-expressions), but it didn't work out.

Reply:
This is probably fine; it's not that expensive and this is not a super common operation.